packages/contracts/contracts/registry/InterfaceBasedRegistry.sol

```diff
@@ -4,55 +4,59 @@ pragma solidity 0.8.10;
-    bytes32 public constant UPGRADE_ROLE = keccak256("UPGRADE_ROLE");
+/// @notice An [ERC-165](https://eips.ethereum.org/EIPS/eip-165)-based registry for contracts
+abstract contract InterfaceBasedRegistry is DAOPermissioned, UUPSUpgradeable {
+    /// @notice The ID of the permission required to call the `_authorizeUpgrade` function.
+    bytes32 public constant UPGRADE_PERMISSION_ID = keccak256("UPGRADE_PERMISSION");
 
+    /// @notice The [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID that the target contracts being registered must support.
     bytes4 public targetInterfaceId;
 
+    /// @notice The mapping containing the registry entries returning true for registererd contract addresses.
     mapping(address => bool) public entries;
 
-    /// @notice Thrown if the contract is already registered
-    /// @param registrant The address of the contract to be registered
+    /// @notice Thrown if the contract is already registered.
+    /// @param registrant The address of the contract to be registered.
     error ContractAlreadyRegistered(address registrant);
 
-    /// @notice Thrown if the contract does not support the required interface
-    /// @param registrant The address of the contract to be registered
+    /// @notice Thrown if the contract does not support the required interface.
+    /// @param registrant The address of the contract to be registered.
     error ContractInterfaceInvalid(address registrant);
 
-    /// @notice Thrown if the address is not a contract
-    /// @param registrant The address of the contract to be registered
+    /// @notice Thrown if the address is not a contract.
+    /// @param registrant The address of the contract to be registered.
     error ContractAddressInvalid(address registrant);
 
-    /// @notice Thrown if the contract do not support ERC165
-    /// @param registrant The address of the contract
+    /// @notice Thrown if the contract do not support ERC165.
+    /// @param registrant The address of the contract.
     error ContractERC165SupportInvalid(address registrant);
 
-    /// @notice Initializes the component
-    /// @dev This is required for the UUPS upgradability pattern
-    /// @param _managingDao The interface of the DAO managing the components permissions
-    /// @param _targetInterfaceId The ERC165 interface id of the contracts to be registered
+    /// @notice Initializes the component.
+    /// @dev This is required for the UUPS upgradability pattern.
+    /// @param _managingDao The interface of the DAO managing the components permissions.
+    /// @param _targetInterfaceId The [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface id of the contracts to be registered.
     function __InterfaceBasedRegistry_init(IDAO _managingDao, bytes4 _targetInterfaceId)
         internal
         virtual
         onlyInitializing
     {
-        __Permissions_init(_managingDao);
+        __DAOPermissioned_init(_managingDao);
```

Comment:
I prefer the original name

Reply:
Duplicate of https://github.com/aragon/core/pull/62#discussion_r928890402 .

Reply:
Let's discuss this here https://github.com/aragon/core/pull/62#discussion_r928957492 and close this conversation.

Reply:
Stated above. Like the old way.